compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEFuseProducerIntoLoop.cpp

```diff
@@ -143,24 +151,31 @@ void AMDAIEFusePackIntoLoopPass::runOnOperation() {
         }
       }
 
-      // Case where operand of generic is a pack op which is in a different
-      // block than the generic's block.
+      // Case where operand of a generic op is a pack/copy op which is in a
+      // different block than the generic's block.
       else if (auto parent = dyn_cast_if_present<tensor::PackOp>(
                    operand.getDefiningOp())) {
         Block *genericBlock = genericOp->getBlock();
         if (parent->getBlock() != genericBlock && parent->hasOneUse()) {
           Operation *firstOpInBlock = &genericBlock->front();
           rewriter.moveOpBefore(parent, firstOpInBlock);
         }
+      } else if (auto parent = dyn_cast_if_present<linalg::CopyOp>(
+                     operand.getDefiningOp())) {
+        Block *genericBlock = genericOp->getBlock();
+        if (parent->getBlock() != genericBlock && parent->hasOneUse()) {
+          Operation *firstOpInBlock = &genericBlock->front();
+          rewriter.moveOpBefore(parent, firstOpInBlock);
+        }
```

Comment:
This can be merged into the previous `if` itself as `parent` is being used as a generic operation.
```
if (isa_and_present<tensor::PackOp, linalg::CopyOp>(operand.getDefiningOp())) {
  Operation* parent = operand.getDefiningOp();
  . . .
}
```